src/query/ast/src/parser/expr.rs

```diff
@@ -827,34 +830,97 @@ pub fn expr_element(i: Input) -> IResult<WithSpan<ExprElement>> {
             }
         },
     );
+
+    let window_frame_bound_start = alt((
+        value(WindowFrameBound::CurrentRow, rule! { CURRENT ~ ROW }),
+        map(rule! { #subexpr(0) ~ PRECEDING }, |(expr, _)| {
+            WindowFrameBound::Preceding(Some(Box::new(expr)))
+        }),
+        value(
+            WindowFrameBound::Preceding(None),
+            rule! { UNBOUNDED ~ LEADING },
+        ),
+    ));
+
+    let window_frame_bound_end = alt((
+        value(WindowFrameBound::CurrentRow, rule! { CURRENT ~ ROW }),
+        map(rule! { #subexpr(0) ~ FOLLOWING }, |(expr, _)| {
+            WindowFrameBound::Following(Some(Box::new(expr)))
+        }),
+        value(
+            WindowFrameBound::Following(None),
+            rule! { UNBOUNDED ~ FOLLOWING },
+        ),
+    ));
+
+    let window_spec = map(
+        rule! {
+            (PARTITION ~ ^BY ~ #comma_separated_list1(subexpr(0)))?
+            ~ ( ORDER ~ ^BY ~ ^#comma_separated_list1(order_by_expr) )?
+            ~ ((ROWS | RANGE) ~ BETWEEN ~ #window_frame_bound_start ~ AND ~ #window_frame_bound_end)?
```

Comment:
`PARTITION BY xxx ORDER BY yyy ROWS UNBOUNDED PRECEDING` is legal, so `BETWEEN` and `AND` should be optional.